benchmark/test/test_httpapi.py

```diff
@@ -0,0 +1,123 @@
+from json import dumps, loads
+
+from twisted.application.internet import StreamServerEndpointService
+from twisted.internet import reactor, endpoints
+from twisted.internet.defer import Deferred, succeed
+from twisted.internet.endpoints import TCP4ServerEndpoint
+from twisted.web import server, client
+from twisted.web.iweb import IBodyProducer
+from twisted.trial import unittest
+
+from zope.interface import implementer
+
+from benchmark.httpapi import BenchmarkAPI_V1, InMemoryBackend
+
+
+@implementer(IBodyProducer)
+class StringProducer(object):
+    def __init__(self, body):
+        self.body = body
+        self.length = len(body)
+
+    def startProducing(self, consumer):
+        consumer.write(self.body)
+        return succeed(None)
+
+    def pauseProducing(self):
+        pass
+
+    def stopProducing(self):
+        pass
+
+
+class TestEndpoint(TCP4ServerEndpoint):
+    def __init__(self, reactor, deferred):
+        super(TestEndpoint, self).__init__(reactor, 0, interface='127.0.0.1')
+        self.deferred = deferred
+
+    def listen(self, protocolFactory):
+        d = super(TestEndpoint, self).listen(protocolFactory)
+
+        def invoke_callback(listening_port):
+            self.deferred.callback(listening_port)
+            return listening_port
+
+        d.addCallback(invoke_callback)
+        return d
+
+
+class BenchmarkAPITest(unittest.TestCase):
```

Comment:
I decided to use the `testtools` library directly. I am making the suggested changes. Additionally, I had to provide `timeout` to `AsynchronousDeferredRunTest`, because the default one of 0.005 seconds was not enough to test HTTP requests over TCP/IP.